### ISSUE TITLE
Bug: when deleting a folder that contains a file we can't delete, we get an uninformative message

### ISSUE THREAD

**Issue by imsnif** [imsnif/diskonaut#11]
To reproduce:
1. Create folder `/path_to_my/folder`
2. Create nested_folder `/path_to_my/folder/nested_folder`
3. Create a file inside nested_folder `/path_to_my_folder/folder/nested_folder/file.txt`
4. Change the permissions of `nested_folder` to be read only.
5. Start diskonaut and give it `/path_to_my` as its path (`diskonaut /path_to_my`).
6. Try to delete folder.
7. See that we get an uninformative error message and are unsure what happened.

In this scenario, the message should read something like "Permission denied deleting /path_to_my/folder/nested_folder".

**Comment by u5surf** [comment]
@imsnif 
> Try to delete folder.
Is this deleting folder `path_to_my` or `nested_folder`
I prefer the message that you can figure out which folder can’t you delete.Although, current implementation is not able to delete  the folder and its nested folders one by one recursively. Thus it difficult to indicate which is not permitted folder

**Comment by imsnif** [comment]
I agree @u5surf. We'll have to implement a recursive delete of our own and catch these errors in order to do this. I would also definitely like to have the error showing the folder that failed and not its parent.
